projects/common/src/utilities/formatters/string/highlight.pipe.ts

```diff
@@ -1,20 +1,41 @@
 import { Pipe, PipeTransform } from '@angular/core';
+import { isArray } from 'lodash-es';
 import { assertUnreachable } from '../../lang/lang-utils';
 
 @Pipe({ name: 'htHighlight' })
 export class HighlightPipe implements PipeTransform {
   public transform(
     fullText: string,
-    highlightedText: string | undefined,
+    highlightSnippets: string | undefined | TextHighlightConfig[],
```

Comment:
If highlight type is now provided here, does it make sense to keep the last arg? The highlightConfig is a superset of any other config, so leaving the string there makes sense as a simple default but any customization of mark can rely on the more complex api. Could also potentially accept a single `TextHighlightConfig` if you want to disambiguate.

Reply:
combining this along with this comment (https://github.com/hypertrace/hypertrace-ui/pull/424#discussion_r540938447), I think it makes sense to keep the arg in case the whole string needs to be highlighted.

Reply:
When designing an API, one of the most important things is making it have consistent, predictable behavior. That is, arguments should always make sense independently and never conflict or require each other (as in, if I optionally took one argument, I shouldn't require another optional argument to also be provided)

Here, it allows me to pass `text, undefined, <mark>` which has clear behavior expectations but is awkward to use (because you have to skip an optional arg to get to a different optional arg). If I pass `text, <config>, <mark>` it's unclear which mark will be used - the one from the config or the one from the mark. If we removed the mark arg entirely, every use case is still satisfied. I could highlight the whole string with `<text>, { <text>, <mark> }` - a bit awkward too with the repetition of text, but with clearly predictable behavior. Omitting the second arg entirely is just a syntax shortcut - but still reachable behavior when using the full api.